### DISCUSSION TITLE
--target @next for specific packages

### DISCUSSION THREAD

**Discussion by zaaakher** [raineorshine/npm-check-updates#1411] [Q&A] [answered]
Hi,

Thank you @raineorshine for this wonderful package.

I use the config `.ncurc.json` to reject certain packages from being updated. And I was wondering if there's a way I can dictate a specific target tag for certain packages in particular?

So my team have a package `@sikka/hawa` and we're always using it in the @next tag so we're always testing the latest. We always have that package like this for example
```
dependencies: {
    "@sikka/hawa": "0.34.0-next"
}
```

But when I run the `ncu` command it sometimes updates `@sikka/hawa` to the latest release like:
```
dependencies: {
    "@sikka/hawa": "0.33.0"
}
```

And I have a feeling if I use the flag `--target @next` when I run `ncu` then I might also update other dependencies to their @next tag.

## Suggestion

I'm thinking maybe the `.ncurc` config file can take an option like this maybe? 👇
```
{
  "reject": ["react-router-dom", "styled-components"],
  "target": [
    {
      "name": "@sikka/hawa",
      "tag": "next"
    },
    {
      "name": "other-package",
      "tag": "beta"
    }
  ]
}
```

I was looking at `src/lib/getNcuRc.ts` and from my understanding so far I think the keys in the config file are the same as the cli flags. So maybe it's already implemented I just have to use the "target" key in the config file properly.

**Comment by raineorshine** [comment, marked as answer]
Hi. Yes, this is currently supported in the `.ncurc.js` config. You can define a `target` function like this:

```js
module.exports = {
  target: name => name === '@sikka/hawa' ? '@next' : 'latest'
}
```

https://github.com/raineorshine/npm-check-updates#target

**Reply by zaaakher** [reply]
Yup that's exactly what I'm looking for. Much appreciated, Thanks @raineorshine 🌹
